### ISSUE TITLE
Accuracy is not changing in the model during training

### ISSUE THREAD

**Issue by renardeinside** [tensorflow/recommenders#479]
Hi team, 
IHAC an application that is designed to train models with TFRS. The basic model is here:

```
class BasicModel(Model):
    def __init__(
        self,
        rating_weight: float,
        retrieval_weight: float,
        product_ids: List[str],
        user_ids: List[str],
    ) -> None:
        # We take the loss weights in the constructor: this allows us to instantiate
        # several model objects with different loss weights.

        super().__init__()

        embedding_dimension = 32

        self.product_ids = np.array([p.encode() for p in product_ids])
        self.user_ids = np.array([u.encode() for u in user_ids])

        self.product_ids_dataset = tf.data.Dataset.from_tensor_slices(
            self.product_ids
        ).cache()
        self.user_ids_dataset = tf.data.Dataset.from_tensor_slices(
            self.user_ids
        ).cache()

        # User and movie models.
        self.product_model = Sequential(
            [
                StringLookup(vocabulary=self.product_ids, mask_token=None),
                tf.keras.layers.Embedding(
                    len(self.product_ids) + 1, embedding_dimension
                ),
            ]
        )

        self.user_model = Sequential(
            [
                StringLookup(vocabulary=self.user_ids, mask_token=None),
                tf.keras.layers.Embedding(len(self.user_ids) + 1, embedding_dimension),
            ]
        )

        # A small model to take in user and product embeddings and predict ratings.
        # We can make this as complicated as we want as long as we output a scalar
        # as our prediction.
        self.ranking_model = Sequential(
            [
                tf.keras.layers.Dense(256, activation="relu"),
                tf.keras.layers.Dense(128, activation="relu"),
                tf.keras.layers.Dense(1),
            ]
        )

        # The tasks.
        self.ranking_task: Ranking = Ranking(
            loss=tf.keras.losses.MeanSquaredError(),
            metrics=[tf.keras.metrics.RootMeanSquaredError()],
        )

        self.retrieval_task: Retrieval = Retrieval(
            metrics=FactorizedTopK(
                candidates=self.product_ids_dataset.batch(128).map(self.product_model)
            )
        )

        # The loss weights.
        self.rating_weight = rating_weight
        self.retrieval_weight = retrieval_weight

    def call(self, features: Dict[Text, tf.Tensor]) -> tf.Tensor:
        # We pick out the user features and pass them into the user model.
        user_embeddings = self.user_model(features["user_id"])
        # And pick out the product features and pass them into the product model.
        product_embeddings = self.product_model(features["product_id"])

        return (
            user_embeddings,
            product_embeddings,
            # We apply the multi-layered rating model to a concatenation of
            # user and product embeddings.
            self.ranking_model(
                tf.concat([user_embeddings, product_embeddings], axis=1)
            ),
        )

    def _build_index(self) -> tfrs.layers.factorized_top_k.BruteForce:
        bf_index = tfrs.layers.factorized_top_k.BruteForce(
            self.user_model
        ).index_from_dataset(
            tf.data.Dataset.from_tensor_slices(self.product_ids)
            .batch(100)
            .map(lambda product_id: (product_id, self.product_model(product_id)))
        )
        # we need to trigger index at least once to make it serializable
        _ = bf_index(np.array(self.user_ids)[[0]])
        return bf_index

    @property
    def index(self) -> tfrs.layers.factorized_top_k.BruteForce:
        return self._build_index()

    def compute_loss(
        self, features: Dict[Text, tf.Tensor], training=False
    ) -> tf.Tensor:
        ratings = features.pop("rating")

        user_embeddings, product_embeddings, rating_predictions = self.call(features)

        # We compute the loss for each task.
        rating_loss = self.ranking_task(
            labels=ratings,
            predictions=rating_predictions,
        )
        retrieval_loss = self.retrieval_task(user_embeddings, product_embeddings)

        # And combine them using the loss weights.
        return self.rating_weight * rating_loss + self.retrieval_weight * retrieval_loss
```

I've tried to train this model with different batch_size, learning rate etc, but all the results are showing that the accuracy metric is not changing at all:
```
13/13 [==============================] - 1s 54ms/step - root_mean_squared_error: 0.4837 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.8863 - regularization_loss: 0.0000e+00 - total_loss: 1059.8863 - val_root_mean_squared_error: 0.3402 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.7792 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.7792
Epoch 10/20
13/13 [==============================] - 1s 51ms/step - root_mean_squared_error: 0.5042 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.9246 - regularization_loss: 0.0000e+00 - total_loss: 1059.9246 - val_root_mean_squared_error: 0.3778 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.8062 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.8062
Epoch 11/20
13/13 [==============================] - 1s 53ms/step - root_mean_squared_error: 0.4653 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.8751 - regularization_loss: 0.0000e+00 - total_loss: 1059.8751 - val_root_mean_squared_error: 0.3645 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.7963 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.7963
Epoch 12/20
13/13 [==============================] - 1s 49ms/step - root_mean_squared_error: 0.5355 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.9370 - regularization_loss: 0.0000e+00 - total_loss: 1059.9370 - val_root_mean_squared_error: 0.3359 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.7764 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.7764
Epoch 13/20
13/13 [==============================] - 1s 50ms/step - root_mean_squared_error: 0.4987 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.9206 - regularization_loss: 0.0000e+00 - total_loss: 1059.9206 - val_root_mean_squared_error: 0.3185 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.7649 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.7649
Epoch 14/20
13/13 [==============================] - 1s 49ms/step - root_mean_squared_error: 0.4883 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.9024 - regularization_loss: 0.0000e+00 - total_loss: 1059.9024 - val_root_mean_squared_error: 0.3664 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.7977 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.7977
Epoch 15/20
13/13 [==============================] - 1s 49ms/step - root_mean_squared_error: 0.4761 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.8845 - regularization_loss: 0.0000e+00 - total_loss: 1059.8845 - val_root_mean_squared_error: 0.3220 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.7671 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.7671
Epoch 16/20
13/13 [==============================] - 1s 49ms/step - root_mean_squared_error: 0.5217 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.9252 - regularization_loss: 0.0000e+00 - total_loss: 1059.9252 - val_root_mean_squared_error: 0.3715 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.8015 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.8015
Epoch 17/20
13/13 [==============================] - 1s 52ms/step - root_mean_squared_error: 0.4989 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.9242 - regularization_loss: 0.0000e+00 - total_loss: 1059.9242 - val_root_mean_squared_error: 0.3499 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.7859 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.7859
Epoch 18/20
13/13 [==============================] - 1s 54ms/step - root_mean_squared_error: 0.5171 - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 1059.9339 - regularization_loss: 0.0000e+00 - total_loss: 1059.9339 - val_root_mean_squared_error: 0.3579 - val_factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - val_factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - val_loss: 1059.7916 - val_regularization_loss: 0.0000e+00 - val_total_loss: 1059.7916
Epoch 19/20
```

Any advisory on what might be wrong here?

**Comment by renardeinside** [comment]
mvw, I've just forgotten to pass user_id/product_id and rating in the proper order to the fit method generator.
